ste/sender-pageBlob.go

```diff
@@ -220,17 +223,41 @@ func (s *pageBlobSenderBase) Prologue(ps common.PrologueState) (destinationModif
 		if err := blockBlobTier.Parse(string(s.destBlobTier)); err != nil { // i.e it's not block blob tier
 			// Set the latest service version from sdk as service version in the context.
 			ctxWithLatestServiceVersion := context.WithValue(s.jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
-			if _, err := s.destPageBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{}); err != nil {
-				if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
-					s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-					s2sAccessTierFailureLogStdout.Do(func() {
-						glcm := common.GetLifecycleMgr()
-						glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-					})
+
+			// First, let's check if the intended tier is a premium page blob tier.
+			// These aren't available on non-premium storage accounts, but we can still copy the data.
+
+			shouldSet := true
+			if premiumPageBlobTierRegex.MatchString(string(s.destBlobTier)) {
+				infoResp, err := s.destPageBlobURL.GetAccountInfo(ctxWithLatestServiceVersion)
+				if err != nil {
+					// If GetAccountInfo fails, this transfer should fail because we lack at least one available permission
+					// (https://docs.microsoft.com/en-us/rest/api/storageservices/get-account-information#authorization)
```

Comment:
I notice that the doc page only mentions shared key and SAS auth. For that reason, can you please test it with OAuth and note the result here.

Reply:
It fails. I have added an alternate code path in case the user is potentially unable to obtain the account details. Even if set tier fails, the transfer will still succeed via OAuth.